commit 31a157cc8470cdc44db0ca299c4c246bd54e2d1c
Author: Asad Sajjad Ahmed <asadsa@varnish-software.com>
Date:   Mon Oct 3 16:08:30 2022 +0200

    h2: add vtc for VSV00011
    
    Check for correct handling of missing pseudo-headers, and invalid
    characters.
    
    Signed-off-by: Asad Sajjad Ahmed <asadsa@varnish-software.com>

diff --git a/bin/varnishtest/tests/f00011.vtc b/bin/varnishtest/tests/f00011.vtc
new file mode 100644
index 000000000..810918028
--- /dev/null
+++ b/bin/varnishtest/tests/f00011.vtc
@@ -0,0 +1,51 @@
+varnishtest "H2: Malformed pseudo-headers"
+
+server s1 {
+	rxreq
+	txresp
+} -start
+
+varnish v1 -arg "-p feature=+http2" -vcl+backend {
+} -start
+
+client c1 {
+	stream 1 {
+		txreq -url ""
+		rxrst
+	} -run
+} -run
+
+client c1 {
+	stream 1 {
+		txreq -url " \t"
+		rxrst
+	} -run
+} -run
+
+client c1 {
+	stream 1 {
+		txreq -scheme ""
+		rxrst
+	} -run
+} -run
+
+client c1 {
+	stream 1 {
+		txreq -scheme " \t"
+		rxrst
+	} -run
+} -run
+
+client c1 {
+	stream 1 {
+		txreq -req ""
+		rxrst
+	} -run
+} -run
+
+client c1 {
+	stream 1 {
+		txreq -req " \t"
+		rxrst
+	} -run
+} -run
